test/EcdsaXi2023Cryptosuite.spec.js

```diff
@@ -203,8 +309,34 @@ describe('EcdsaXi2023Cryptosuite', () => {
     it('should fail to sign with non-bytes extraInformation', async () => {
       const unsignedCredential = JSON.parse(JSON.stringify(credential));
       unsignedCredential.type.push('UndefinedType');
+      const badXI = 100;
+      const badCryptosuite = createCryptosuite({badXI});
```

Comment:
- This doesn't look like it's doing the intended check.
- The `UndefinedType` above is triggering the "relative type URL" test issue above.  jsonld is throwing a safe mode error.  I think you can remove that `push` since it's from another test?
- After that, the internals here are going to treat `{badXI: 100}` as `{}` since it only looks for `extraInformation`.  I think you'd want `{extraInformation: badXI}`?
- Ideally the code would do deeper checks on the resulting `error`.  Which is kind of annoying at times, but ensures a test isn't passing due to the wrong error happening.
- In these new tests, I think the errors are all `TypeError`.  The 'jsonld.ValidationError' was copy & paste from the safe mode checks above.

Reply:
Good catches, thanks. Fixed.